docker-entrypoint-tcp.sh

```diff
@@ -6,6 +6,10 @@ set -e
 # * if hyphenated flag arguments (e.g., '-f', '-foo', or '--foo') were
 #   passed to the Docker command line
 if [ "$#" -eq 0 ] || [ "${1#-}" != "$1" ]; then
+    export CCACHE_DIR="/var/cache/ccache"
```

Comment:
`CCACHE_DIR` should ideally be configurable, i.e., something along the lines of `CCACHE_DIR="${CCACHE_DIR:-/var/cache/ccache}"`

Reply:
That is the default location which gentoo would use for its cache if you follow the docs. And note that here we are talking about "inside" the container so the location does not matter too much. Why do you think users might want to control that location?

env variables are kind of nasty to get into containers, after which they can get confusing especially if the same key has different values on host and in chroot. But i guess people relying on containers might have learned that by now.

Reply:
There is just no reason to hardcode it. I don't think a user would ever want to touch it, but should they need to, they should be able to.

On second thought, the proper place to define it would be in the Dockerfile using an `ENV CCACHE_DIR=/var/cache/ccache` directive. Assuming of course that this PR needs to access this envvar; if not, it should not be defined at all.

Reply:
Putting it into every env could be useful to deal with the ssh image and to shorten the call for the stats. Will look into that.

Reply:
In fact the variable `CCACHE_DIR` is not needed at all. `ccache` has its defaults and will create a directory on demand, also no config file needed.

As soon as the PATH is prepended `ccache` is used. Can even be used without mounting in anything, however then the cache will only live as long as the container.

Users should probably mount to the default location, or to anywhere while also setting `CCACHE_DIR` as they want. I will have to look into that `VOLUME` command for the Dockerfile to see how that interacts.

Reply:
Ok i decided to go with `CCACHE_DIR` as the "master enable" switch for both ssh and tcp. In fact it turned out they are required in both because defaults do not work.

In the case of tcp the default does not work because the user 'distcc' does not have a HOME, and distccd changes to that user.